### PR TITLE
Add links to all issues in the 3.7.1 and 3.7.0 sections of the changelog

### PR DESCRIPTION
**Pull request by rodrigoprimo**
## Description

This PR adds links to all issues in the 3.7.1 and 3.7.0 sections of the changelog. I also fixed a typo in a changelog entry.

## Related issues/external references

Part of #244 

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [x] Documentation improvement

## PR checklist

- [x] I have checked there is no other PR open for the same change.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [x] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [ ] I have added tests to cover my changes.
- [ ] I have verified that the code complies with the projects coding standards.
- [ ] [Required for new sniffs] I have added XML documentation for the sniff.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -403,24 +405,35 @@ Additionally, thanks to [Alexander Turek][@derrabus] for consulting on the repo
...
+- Fixed bug [#3604]: Tokenizer/PHP: bug fix for double quoted strings using ${
     - Thanks to [Juliette Reinders Folmer][@jrfnl] for the patch
 
+[#3502]: https://github.com/squizlabs/PHP_CodeSniffer/issues/3502
+[#3503]: https://github.com/squizlabs/PHP_CodeSniffer/issues/3503
+[#3505]: https://github.com/squizlabs/PHP_CodeSniffer/issues/3505
+[#3526]: https://github.com/squizlabs/PHP_CodeSniffer/issues/3526
+[#3530]: https://github.com/squizlabs/PHP_CodeSniffer/issues/3530
+[#3534]: https://github.com/squizlabs/PHP_CodeSniffer/issues/3534
+[#3546]: https://github.com/squizlabs/PHP_CodeSniffer/issues/3546
```
Comment (jrfnl) [inline CHANGELOG.md:432]: These are PRs, not issues.

File: CHANGELOG.md [hunk 2]
```diff
@@ -403,24 +405,35 @@ Additionally, thanks to [Alexander Turek][@derrabus] for consulting on the repo
...
+[#3550]: https://github.com/squizlabs/PHP_CodeSniffer/issues/3550
+[#3575]: https://github.com/squizlabs/PHP_CodeSniffer/issues/3575
+[#3604]: https://github.com/squizlabs/PHP_CodeSniffer/issues/3604
```
Comment (jrfnl) [inline CHANGELOG.md:435]: Dito.

File: CHANGELOG.md (hunk 1 above)
Comment (rodrigoprimo) [reply CHANGELOG.md]: Sorry about that. I missed it when testing the links, as GitHub redirects to the correct URL. It is fixed now. I opted to amend the original commit as this was a small documentation fix.

File: CHANGELOG.md (hunk 2 above)
Comment (rodrigoprimo) [reply CHANGELOG.md]: Same as above

File: CHANGELOG.md (hunk 1 above)
Comment (jrfnl) [reply CHANGELOG.md]: Understood and yes, GH redirects, but we should not rely on that.

**Review by jrfnl — APPROVED** [review]
Thanks @rodrigoprimo !

**Pull request merged**